File: sprint3/Java/lesson4code1.java
public class Main {
    public static void main(String[] args) {
        genBinary(3, "");
    }

    static void genBinary(int n, String prefix) {
        if (n == 0) {
            System.out.println(prefix);
        } else {
            genBinary(n - 1, prefix + "0");
            genBinary(n - 1, prefix + "1");
        }
    }
}

File: sprint3/By lessons/lesson4code1/lesson4code1.go
package main

import "fmt"

func main() {
    genBinary(3, "")
}

func genBinary(n int, prefix string) {
    if n == 0 {
        fmt.Println(prefix)
    } else {
        genBinary(n-1, prefix+"0")
        genBinary(n-1, prefix+"1")
    }
}
